frontend/packages/operator-lifecycle-manager/src/components/descriptors/status/index.tsx

```diff
@@ -178,69 +114,46 @@ const MainStatus: React.SFC<StatusCapabilityProps> = ({
   );
 };
 
-const capabilityComponents = ImmutableMap<
-  StatusCapability,
-  React.ComponentType<StatusCapabilityProps>
->()
-  .set(StatusCapability.podStatuses, PodStatuses)
-  .set(StatusCapability.conditions, StatusConditions)
-  .set(StatusCapability.w3Link, Link)
-  .set(StatusCapability.k8sPhase, K8sPhase)
-  .set(StatusCapability.k8sPhaseReason, K8sPhaseReason)
-  .set(StatusCapability.k8sResourcePrefix, K8sResourceLink)
-  .set(StatusCapability.hidden, null);
+export const StatusDescriptorDetailsItem: React.FC<StatusCapabilityProps> = (props) => {
+  const capability = _.get(props.descriptor, ['x-descriptors', 0], null) as StatusCapability;
```

Comment:
nit: prefer optional chaining

```suggestion
  const capability = props.descriptor?.['x-descriptors']?.[0] as StatusCapability;
```